data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/ProcessWorker.java

```diff
@@ -117,9 +117,8 @@ private void doRun() {
                 processAcknowledgements(inputEvents, records);
             }
         }
-        if (!records.isEmpty()) {
```

Comment:
Why remove this check?

Reply:
To ensure data does not get stuck in the OpenSearch sink when a flush time is specified and no further traffic comes in. We have to call sink::output to check the flush time which wouldn't happen with this `if` if no traffic ever comes back in